seacrowd/sea_datasets/weathub/weathub.py

```diff
@@ -0,0 +1,143 @@
+"""
+
+_HOMEPAGE = "https://huggingface.co/datasets/iamshnoo/WEATHub"
+
+_LANGUAGES = ["tha", "tgl", "vie", "cmn", "eng"]
+
+_LICENSE = Licenses.CC_BY_4_0.value
+
+_LOCAL = False
+
+_URLS = {
+    "original_weat": "https://huggingface.co/datasets/iamshnoo/WEATHub/resolve/main/data/original_weat-00000-of-00001-7e85a83920777d87.parquet",
+    "new_human_biases": "https://huggingface.co/datasets/iamshnoo/WEATHub/resolve/main/data/new_human_biases-00000-of-00001-557150d939938864.parquet",
+    "india_specific_biases": "https://huggingface.co/datasets/iamshnoo/WEATHub/resolve/main/data/india_specific_biases-00000-of-00001-a2b8ed74878afc1b.parquet",
+}
+
+_SUPPORTED_TASKS = [Tasks.WORD_LIST]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class WEATHubDataset(datasets.GeneratorBasedBuilder):
+    """WEATHub is a dataset containing 24 languages. It contains words organized into groups of (target1, target2, attribute1, attribute2) to measure the association target1:target2 :: attribute1:attribute2. This dataset corresponds to the data described in the paper "Global Voices, Local Biases: Socio-Cultural Prejudices across Languages" accepted to EMNLP 2023."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+    SEA_LANGUAGES = set(["th", "tl", "vi", "zh", "en"])
+
+    subsets = [
+        f"{_DATASETNAME}.original_weat",
+        f"{_DATASETNAME}.new_human_biases",
+        f"{_DATASETNAME}.india_specific_biases",
+    ]
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{sub}_source",
+            version=datasets.Version(_SOURCE_VERSION),
+            description=f"{sub} source schema",
+            schema="source",
+            subset_id=f"{sub}",
+        )
+        for sub in subsets
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}.original_weat_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "language": datasets.Value("string"),
+                    "weat": datasets.Value("string"),
+                    "attr1.category": datasets.Value("string"),
+                    "attr1.examples": [datasets.Value("string")],
+                    "attr2.category": datasets.Value("string"),
+                    "attr2.examples": [datasets.Value("string")],
+                    "targ1.category": datasets.Value("string"),
+                    "targ1.examples": [datasets.Value("string")],
+                    "targ2.category": datasets.Value("string"),
+                    "targ2.examples": [datasets.Value("string")],
+                }
+            )
+        else:
+            raise NotImplementedError()
```

Comment:
I assume there's no seacrowd config here, but perhaps let's raise an appropriate error? I suggest disentangling errors because there's no seacrowd schema and errors where the schema is actually wrong. I'm thinking something like:

```python
if self.config.schema == "source":
    # do something
elif "seacrowd" in self.config.schema:
   raise NotImplementedError("No seacrowd schema because ..."
else
   raise ValueError("Invalid schema name")
```

Reply:
Okay, I'll change it. Thanks for the feedback!